### ISSUE TITLE
Too strong focus on Workshoppers

### ISSUE THREAD

**Issue by martinheidegger** [nodeschool/organizers#378]
Hello everyone.
I have been going through the response of [The missing hands at NodeSchool](https://medium.com/@leichtgewicht/the-missing-hands-at-nodeschool-8999a90d33d1#.g2h9liz6o) and somehow it seems to me that many people think `NodeSchool == Workshoppers`. From my point of view this slightly skewed since there have been presentations of various kinds at nodeschool events. I think it might be worth to change the NodeSchool homepage to show That the workshoppers are resources that you **can** use, and extend the section with useful videos, tutorials, books or other kind of learning materials that just simply made our events better?

**Comment by maxkorp** [comment]
Prefacing this, I haven't been particularly active with nodeschool directly, but I do a lot of educating in AZ, CA, and UT about node, electron, git and other related things.

It definitely felt that way (that `NodeSchool == Workshoppers` was prescribed) when I signed up around 2 years ago. 

I think people also tend to flock to that, as most people don't know where to start with pedagogical endeavors. There's a whole lot of idea soup the first time someone sits down to say "if you want to learn js/programming, you have to learn... tons of stuff", and the workshoppers definitely provide some easy focus to that.

Your suggestions of tutorials, books etc are definitely good (I'd say necessary) and help somewhat, but I think that alone would only extend the issue from `NodeSchool == Workshoppers` to `NodeSchoole == [Workshoppers, Tutorials, etc]`.  To build on those even further, helping organizers understand what, how, and _why_ to teach certain things is critical, because that provides them maximum insight on how to improve events and help people out. Something like an outline of topics, coupled with associated workshoppers, videos, tutorials, etc, would help a ton I think. Perhaps even being able to factor out outlines (almost like bullet-point lesson plans) from associated workshoppers would be great. Maybe something like "frequent questions asked during workshopper X and various explanations" would be amazing. Honestly, such a thing would work nicely as a tutorial analogue to a workshopper. 

That all said, while I don't think workshoppers should be _the_ central focus as they are (at least commonly percieved to be) currently, I think they should stay _a_ central focus. 
I've been doing nodeschool-esque things for a few years now and I still get caught off guard in groups sometimes; It can be hard to keep a group of people wrangled in manually, even with a plan, and having something hardcoded like a workshopper can help with that. 

Sorry for the wall of text, that was way more than I intended, but I think it all stands.

**Comment by martinheidegger** [comment]
No problem about a lot of text. I write much as well. Its the emojis. I am of the immediate feeling that workshoppers should be part of a page called "NodeSchool Study Material". Where the workshoppers are listed as well as online tutorials, etc. It could be parallel to a "NodeSchool Event Material" that hosts facts, infos etc. about hosting Node events.

**Comment by pcuci** [comment]
Many cities have js-meetups already active. Also FreeCodeCamp and
LadiesLearningCode groups exist.

NodeSchool is and can stay different.

Different how?

**Comment by martinheidegger** [comment]
##### `js != nodeJs`

As for the content: A strong focus on Node might be a good as a lock-in. Node is based on JavaScript, and there can be JavaScript Workshoppers in there. Even browser-facing tutorials are okay but i would restrict it to those that have at least touching points with Node.js ー i.e. a workshopper about https://fullcalendar.io/ seems sort of misplaced.

_(Im am aware that this rule would exclude [web-audio-school](), [shader-school](https://github.com/stackgl/shader-school) and [webgl-workshop](https://github.com/mmckegg/web-audio-school) ... not sure how I feel about that ...)_

The workshoppers certainly are a good start but how about Videos? Presentations? News?

As for the events:
My current definition is like
 - Entirely open (If your event is hosted in a place where people can not sign up, its out)
 - Freeish (Access to the event should be free or really affordable - affordable is different per country; Exception: NodeSchool as part of conferences)
 - Focus on Node (See the content section)

**Comment by martinheidegger** [comment]
Note: I pushed this discussion by creating a PR that removes the non-node workshoppers from the website.

**Comment by llkats** [comment]
I gave a [talk in September](https://llkats.github.io/node-interactive-eu-2016-talk/) that in part covered the origins and history of NodeSchool as an organization. The tl;dr that I'd like to point out is that NodeSchool arose directly from the development of workshoppers. Workshoppers were the learning materials that were reused and made these recurring events possible. NodeSchool's decentralized nature means that chapters can use this material to base their events around, and then decide on their own the best ways to support their communities, whatever that means for them. I don't think the format is prescribed, merely suggested, and that's a good thing.

I don't think there's too much of a focus on workshoppers. Having a high-quality curriculum of learning materials at hand means that events don't have to spend time developing this curriculum, and can instead focus on hosting their events. Adding even more material like videos and tutorials is a laudable goal, but it seems to me like there's more than enough work to do updating the material that we have for now.

As far as removing the non-Node workshoppers from the site -- I strongly feel that those workshoppers should not be removed entirely, but rather put into another section, underneath the "core" and "extracurricular" sections, under perhaps a "general interest" or "other programming topics" heading.

**Comment by martinheidegger** [comment]
> I don't think there's too much of a focus on workshoppers.

I would like to Note that I didn't start this thread on a feeling. I have been told by at least persons that they thought the NodeSchool to be synonymous with "studying workshoppers". Which, like you mentioned in your talk, it isn't.

> there's more than enough work

Note that I didn't suggest writing all those stuff, Simple links and giving people the possibility to add the materials they use could be enough.

> I strongly feel that those workshoppers should not be removed entirely

Is there a rationale to that feeling? I do understand that those tutorials have history with nodeschool, but that is the extent that I can see. They will not make NodeSchool events more focussed on Node.js but they do confuse people that attend. Not having a clearly understandable curriculum hurts onboard and this is imho. a part of it.

**Comment by llkats** [comment]
I'm sorry my opinions were not good enough for you. Please disregard.

**Comment by jedireza** [comment]
Git-It for example, not being a "Node JS" work-shopper, has been a huge success helping people learn a really important set of skills.

Besides, weren't these all written in node and are installed using npm? Re-categorizing them makes the most sense, removing them doesn't.

**Comment by martinheidegger** [comment]
Okay, not getting to a green leaf here. Closing this issue.

**Comment by isaacs** [comment]
It seems like y'all're coming to a resolution on this, so maybe I'm too late for this to matter, but I want to add my own thoughts to the points that @llkats made, because she's 100% right.

NodeSchool isn't about "studying workshoppers" any more than a calculus class is about "studying graphs" or literature class is about "looking at words".  The tool is not the output, but before we start abandoning the tool, it's important to take stock of _why_ it works as well as it does.  (And, if you've seen it happen, then you know that it does!)

The benefit of using workshoppers is that they provide a simple and straightforward way for newcomers to get started.  They both _feel small_ and _ask a lot_ of new users.  This is super important.

There is a very discrete hurdle to get over in order to start doing anything with workshoppers.

1. Get Node installed
2. Find the command line (Terminal.app or cmd or whatever)
3. Install something with npm
4. Run a command on the terminal

Once you're there, you can start working.  The rest of the context (from literally the very very beginning of how to use JavaScript and write very basic programs in Node.js), is taught to you as you need it.

Once you're there, you can _help your neighbor_, which reinforces that this is _a skill you have learned_, with _immediate social value_.  Every workshopper you complete also gives you skills that can let you help those around you.  Even more important, along the way, you are forced to interact with this persnickety machine, _with your actual physical hands_, which teaches debugging, patience, and muscle memory.

If everyone in the room isn't doing workshoppers, and instead is watching a talk, or a video, or reading a textbook, or any other thing, you lose those benefits.

Watching a talk is only valuable if you have the _right_ level of context to make the talk make sense.  If you have too _much_ context, it's too elementary.  Too little, and it's just gibberish.  This is why schools break things up by grades and skill levels.  But nodeschool is too small for that.  We need a system that is personalized, but scalable.

The workshopper is the cornerstone of the NodeSchool experience.  It's what makes it work.  You can't lean over and ask the stranger next to you for help with a blog post or a video you're watching.  People definitely look at blog posts and videos at node school.  There are presentations sometimes.  But those things are like the pizza we order.  It's not the _point_ of the event in the way that the workshoppers are.

Furthermore, having workshoppers for things that are not directly related to Node.js (or even to JavaScript) is also perfectly fine.  The most valuable thing that I learned in studying computer science was not anything that any professor said in any class, or even anything that I read in any book, or had to implement in any program.  The most valuable thing I learned was the skill of banging away at computers, adding `printf` statements in programs, and debugging stuff that made no sense to me until it started to make sense to me, as I gradually formed a mental model of what was going on in a computer program.  You can't _tell_ someone that, any more than you can _tell_ someone how to throw a baseball or ride a bike or sing a song.  It requires that you grow new parts in your brain, which is very quickly and effectively done in a social situation with a focused task at hand.

Everyone who does workshoppers learns valuable implicit lessons about Node, npm, and JavaScript.

If one or two people feels like it's not for them, ok.  No big deal.  Every time I go to a node school event (which, sadly, I haven't been able to make for the last few months, because of a new baby), I have met a handful of people who have never programmed before, maybe have never even seen a command line before, and walk out feeling like they've opened up a whole new door in a friendly environment.  It's like they learned a magic trick.  Or they don't, and they decide, "Ok, so that's what that is, I don't like it", and they move on, and _that's cool too_.  Or they're a veteran programmer, and just show up to socialize and help people debug stuff.  (Note: this last one is me.  I don't actually do workshoppers.  I think they're kinda boring.  But I'm not a node newcomer, am I?)

If anything, I think that the magic and power of workshoppers _isn't highlighted enough_ on the website.

@martinheidegger If you are finding that your nodeschool events are not focused or productive enough, maybe that's a problem to focus directly on.  Perhaps your local community would benefit from a different structure?  Or perhaps there are ways that the events could be managed that would give people a little bit more direction?  Once the cultural ball is rolling, and everyone has the expectation about how it's going to go, it's very focusing, and can get everyone aligned in a very powerful way.

**Comment by martinheidegger** [comment]
@isaacs _(Answering since you mentioned me, I closed this issue since I don't want to waste more time on it)_ This issue was originally opened not because I think it is too focussed on workshoppers but because people told me this is a "workshopper organization" and I wanted to mitigate that. Don't want to anymore.

**Comment by mikeal** [comment]
Before I address the specific topics of contention I'd like to make sure we get an important part of the history right.

NodeSchool is an *accident*. There was never any grand plan, we all stumbled upon things that seemed to really work and then built on them. What gets lost in the history is all the things that **didn't** work that many people tried and left behind after sober reflection.

There was a lot of work done, particularly by @jlord and @maxogden, to create a more sustainable organizing structure and website that pushes people towards practices that we've found work. At this point, I think most of the people using these practices don't understand *why* they work and we probably don't do a good job of explaining it :)

Back to the topic at hand.

To me, the key to NodeSchool is the **support system**.

Whether you work through the materials at home or at an in-person meetup there is a way for you to ask questions and get help. That support makes the process of learning sticky, especially when the local repos are active.

At a larger meta level that support system also levels people up into mentors and organizers, which is the key to NodeSchool's long term sustainability.

Workshoppers are key because they drastically lower the bar to becoming a contributor to the support system rather than just a consumer of it. As soon as someone finishes a single challenge they can support the person next to them. Once someone completes half a workshopper they might be able to be a mentor. Once a couple people finish a workshopper online they can start a new chapter in their town.

Workshoppers are an amazing tool in lowering barriers to entry for *mentors* and *organizers*. We spend a lot of time talking about and thinking about how they help learning but the key success to the growth and sustainability has been how easy they make it to be mentors and organizers.

I've had the privilege of visiting a *lot* of NodeSchools around the world. Some at conferences and other events and some in the regular rotation of local meetups. The success of the NodeSchool chapters I've visited can be directly tied to how well they build, sustain, and grow their support system. The ones that naturally bring people into the support system and enable more mentors and organizers are the ones that outgrow their spaces, end up with waiting lists, and where the organizers end up being good friends for life (or at least for the last 4 years or so).

It's a beautiful thing when it all works.

The best thing we can do is share our stories the stories from our community about being organizers and mentors. The next round of iteration should be practices that help to build and sustain local support systems, based on these stories. Or at least that's what I think ;)
